types/firestore.d.ts

```diff
@@ -390,11 +409,11 @@ declare namespace FirebaseFirestore {
      * @return A Promise that resolves with an array of resulting document
      * snapshots.
      */
-    getAll(
+    getAll<AppModelType, DbModelType extends DocumentData>(
```

Comment:
I had originally meant for the declaration of `getAll()` here to remain pinned to DocumentData (despite the implementation in index.ts being generalized), just like it was before the PR. But now that I see that the implementation indeed preserves the type converter, it seems fine to generalize it here. That being said, we should add some test coverage for calling getAll() with a type converter; otherwise, there is no direct test coverage for calling getAll() with type converters. Maybe consider generalizing getAll() in a separate PR so that the corresponding test coverage can be added in a smaller PR and we can get this one merged.